### PR TITLE
Implement ConnOpenAck datagrain in the ICS Connection spec

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: core/src/ibc/connection_03/manager.rs [hunk 1]
```diff
@@ -122,6 +122,45 @@ impl<'a> Manager<'a> {
...
+            return Err(format!("Invalid connection state expected INIT or TRYOPEN but found {:?}", connection.state))
+        }
+        let expected_connection = ConnectionEnd {
+            state: ConnectionState::TRYOPEN,
+            counterparty_connection_identifier: identifier.clone(),
+            counterparty_prefix: get_commiment_prefix(),
+            client_identifier: connection.counterparty_client_identifier.clone(),
+            counterparty_client_identifier: connection.client_identifier.clone(),
+        };
+        self.verify_connection_state(&connection, proof_height, proof_try, identifier.clone(), &expected_connection);
```
Comment (junha1) [inline core/src/ibc/connection_03/manager.rs]: If this function will be moved to client, how will you retrieve a reference to client?

Comment (majecty) [reply core/src/ibc/connection_03/manager.rs]: `ConnectionEnd` contains the client identifier.

Comment (junha1) [reply core/src/ibc/connection_03/manager.rs]: We concluded that all `verify_...` function will take ID of client.

**Comment by majecty** [comment]
@junha1 I added this commit: https://github.com/CodeChain-io/foundry/pull/180/commits/52cc5fd7a8bd239b7b82ea2b8b5810534a7cc9e8

Other things are barely changed. I changed the type of proof to `Vec<u8>` from `String`. I changed the code that was using `get_kv_store` to use `get_kv_store_mut`.

File: core/src/ibc/connection_03/manager.rs [hunk 2]
```diff
@@ -135,13 +174,19 @@ impl<'a> Manager<'a> {
         &mut self,
         connection: &ConnectionEnd,
         proof_height: u64,
-        proof: CommitmentProof,
+        proof: Vec<u8>,
         connection_identifier: Identifier,
         connection_end: &ConnectionEnd,
     ) -> bool {
+        let proof_dec: CommitmentProof = if let Ok(proof) = rlp::decode(&proof) {
```
Comment (junha1) [inline core/src/ibc/connection_03/manager.rs]: decoding should happen only in the client.

Comment (majecty) [reply core/src/ibc/connection_03/manager.rs]: I think that the whole `verify_connection_state` function will be moved to the light client soon.

Comment (junha1) [reply core/src/ibc/connection_03/manager.rs]: Ah ok. You just wanted to keep that consistent. I'll move it soon, so never mind.

**Pull request merged**